### PR TITLE
Add monthly mode to stats

### LINKED ISSUE #52: As a user who has a limited budget, I would like to view my expenses over a certain period of time
so that I can learn from my past endeavours and better manage my budget

### PR DESCRIPTION
**Pull request by jonathantjm**
resolves #52

### CODE REVIEW

**Comment by JasonChong96** [comment]
Merge conflict

File: src/test/java/seedu/address/logic/LogicManagerTest.java [hunk 1]
```diff
@@ -60,12 +66,39 @@ public void getFilteredExpenseList_modifyList_throwsUnsupportedOperationExceptio
...
+        assertTrue(map.get(validExpense.getDate().toString()) == validExpense.getCost().getCostValue());
+
+        model.updateStatsMode(StatsMode.MONTH);
+        map = logic.getExpenseStats();
+        String month = new SimpleDateFormat("MMM-YYYY").format(validExpense.getDate().fullDate.getTime());
+        assertTrue(map.size() > 0);
+        assertTrue(map.containsKey(month));
+        assertTrue(map.get(month) == validExpense.getCost().getCostValue());
+
+        model.deleteExpense(validExpense);
```
Comment (winsonhys) [inline src/test/java/seedu/address/logic/LogicManagerTest.java]: Dont delete the expense at the end. Use a `@BeforeEach` annotated method instead to reset the model

File: src/test/java/seedu/address/logic/commands/StatsCommandTest.java [hunk 2]
```diff
@@ -5,22 +5,26 @@
...
 import seedu.address.model.exceptions.NonExistentUserException;
 import seedu.address.model.exceptions.UserAlreadyExistsException;
+import seedu.address.testutil.Assert;
 import seedu.address.testutil.ModelUtil;
 import seedu.address.ui.testutil.EventsCollectorRule;
 
 public class StatsCommandTest {
     @Rule
-    public final EventsCollectorRule eventsCollectorRule = new EventsCollectorRule();
+    public final EventsCollectorRule eventsCollectorRule = new EventsCollectorRule(
```
Comment (winsonhys) [inline src/test/java/seedu/address/logic/commands/StatsCommandTest.java]: Does this compile

File: src/test/java/seedu/address/logic/LogicManagerTest.java (hunk 1 above)
Comment (jonathantjm) [reply src/test/java/seedu/address/logic/LogicManagerTest.java]: ok done

File: src/test/java/seedu/address/logic/commands/StatsCommandTest.java (hunk 2 above)
Comment (jonathantjm) [reply src/test/java/seedu/address/logic/commands/StatsCommandTest.java]: no haha. fixed it

**Pull request merged**